devito/builtins.py

```diff
@@ -431,8 +435,13 @@ def sumall(f):
     # otherwise we would eventually be summing more than expected
     p, eqns = f.guard() if f.is_SparseFunction else (f, [])
 
+    s = dv.types.Scalar(name='sum', dtype=f.dtype)
+
     with MPIReduction(f) as mr:
-        op = dv.Operator(eqns + [dv.Inc(mr.n[0], p)], name='sum')
+        op = dv.Operator([dv.Eq(s, 0.0)] +
+                         eqns +
+                         [dv.Inc(s, p), dv.Eq(mr.n[0], s)],
+                         name='sum')
```

Comment:
These two bultins update look very hacky.

Reply:
No it's fine, no devito issues here. it's just to work around what I think to be an openmp 5 (clang) bug

```
#pragma omp ... reduction(+:s)
```

works like a charm

```
#pragma omp ... reduction(+:n[0])
```

does not (ie, with arrays), ie it produces garbage. It works fine without offloading, so...

Reply:
But what do you need `mr` for then? You already have the result in `s` then copy it in `mr.n[0]` then output it, that's a temporary and extra eq that makes it look odd for no specific reason.

Reply:
`mr.[0]` has always been there

it's what you need to propagate a scalar back to the user

you cannot return `s` because it's not passed by reference

Reply:
Hum ok, I get your point. Looks like a quite "unnatural" way to do it, so some user may run into issues trying to add a norm into their own Operator. Food for thoughts